docs/source/how-to/docker.md

```diff
@@ -67,17 +67,128 @@ quickly generate some with:
 python -m tiled.examples.generate_files files/
 ```
 
+### Quick Start (Not Scalable)
+
+This approach is nice for development and rapid iteration. It indexes the files
+at server startup.
+
 ```
 docker run \
   -p 8000:8000 \
   -e TILED_SINGLE_USER_API_KEY=secret \
-  -v ./files:/files:ro ghcr.io/bluesky/tiled:latest \
+  -v ./files:/files:ro \
+  ghcr.io/bluesky/tiled:latest \
   tiled serve directory --host 0.0.0.0 /files
 ```
 
+Two problems with this one-line approach:
+
+* If you restart the server, all the indexing work is done from scratch.
+* If you horizontally scale with multiple containers, each one will crawl the
+  filesystem individually, putting load on the filesystem and potentially getting
+  views of the filesystem that are out of sync.
+
+Read on for a scalable approach.
+
+### Scalable to Multiple Processes
+
+Create a place outside the container to store the "catalog", `catalog.db`.
+
+```
+mkdir storage/
+```
+
+Start the server, potentially multiple on different ports.
+
+```
+docker run \
+  -p 8000:8000 \
+  -e TILED_SINGLE_USER_API_KEY=secret \
+  -v ./files:/files:ro \
+  -v ./storage:/storage \
+  ghcr.io/bluesky/tiled:latest
+```
+
+Register the files in the directory `files/` with the catalog.
+
+```
+docker run \
+  -e TILED_SINGLE_USER_API_KEY=secret \
+  -v ./files:/files:ro \
+  -v ./storage:/storage \
+  ghcr.io/bluesky/tiled:latest \
+  tiled catalog register /storage/catalog.db /files --verbose
+```
+
+### Scalable to Multiple Hosts
+
+Instead of the default SQLite database, we need to use a PostgreSQL database.
+One way to run a PostgresSQL database is:
+
+```
+docker run --name tiled-test-postgres -p 5432:5432 -e POSTGRES_PASSWORD=secret -d docker.io/postgres
+```
+
+Initialize the database. (This creates the tables, indexes, and so on used by Tiled.)
+
+```
+export TILED_DATABASE_PASSWORD=secret
+export TILED_DATABASE_URI=postgresql+asyncpg://postgres:${TILED_DATABASE_PASSWORD}@localhost:5432
+
+docker run --net=host ghcr.io/bluesky/tiled:latest tiled catalog init $TILED_DATABASE_URI
+```
+
+Create a directory for TIled configuration, e.g. `config/`.
+
+```
+mkdir config/
+```
+
+Place a copy of `example_configs/single_catalog_single_user.yml`, from the Tiled
+repository root, in this `config/` directory.
+
+Replace the line:
+
+
+```yaml
+uri: "sqlite+aiosqlite:////storage/catalog.db"
+```
+
+with a PostgreSQL database URI, such as:
+
+```yaml
+uri: "postgresql+asyncpg://postgres:${TILED_DATABASE_PASSWORD}@localhost:5432"
+```
+
+Start the server, potentially multiple across many hosts.
```

Comment:
```suggestion
Start the server, potentially multiple servers across many hosts.
```

Reply:
Eventually, it would be useful to include in a cookbook the example of setting up multiple servers with a shared db.

E.g., does each server need to have the same config settings?